lib/sync/storage/dataset-clients.js

```diff
@@ -436,6 +443,52 @@ function doUpdateManyDatasetClients(query, fields, callback) {
   });
 }
 
+/**
+ * 
+ * Save the record uid mappng in the database 
+ * @param {String} datasetId the datasetId 
+ * @param {String} clientUid the client uid
+ * @param {String} serverUid the server uid
+ * @param {Function} callback
+ */
+function doSaveRecordUidMapping(datasetId, clientUid, serverUid, callback) {
+  debug('doSaveRecordUidMapping datasetId = %s, clientUid = %s, serverUid= %s', datasetId, clientUid, serverUid);
+  if (!clientUid || !serverUid) {
+    debug('invalid clientUid or serverUid. Nothing to save');
+    return callback();
```

Comment:
I don't think so. If any of them is null, we just don't save them. It is possible for both of them to be null, but it is covered here?